keanu-project/src/main/java/io/improbable/keanu/vertices/generic/probabilistic/discrete/CategoricalVertex.java

```diff
@@ -17,7 +21,8 @@
 
 import static java.util.stream.Collectors.toMap;
 
-public class CategoricalVertex<T> extends Vertex<T> implements Probabilistic<T> {
+
+public class CategoricalVertex<T> extends Vertex<GenericTensor<T>> implements Probabilistic<GenericTensor<T>> {
```

Comment:
Do you think a javadoc on how this vertex works with tensors is useful? Or is it clear from context?